packages/labs/react/src/create-component.ts

```diff
@@ -315,7 +317,11 @@ export function createComponent<
         if (
           eventProps.has(k) ||
           (!reservedReactProperties.has(k) &&
-            !(k in HTMLElement.prototype) &&
+            // NODE_MODE check below is to support React SSR where HTMLElement
+            // might not be defined
+            !(NODE_MODE && typeof HTMLElement === 'undefined'
+              ? false
+              : k in HTMLElement.prototype) &&
```

Comment:
I feel like this ternary should be able to be done at the `module` level.

There should be a way to get the `prototype` or `false` outside of the `ReactComponent.render` function and outside of the `createComponent` function?

```
const SERVER_CHECK = <is node server> ? element : HTMLElement;
```

then later we can use the appropriate prototype instead of both?

Reply:
The ternary is potentially run for every property on every render. It should be only called when necessary.

Also the following in-out check disappears in https://github.com/lit/lit/pull/3128
```
 !(k in HTMLElement.prototype) && ...
```

So this check might not conflict with the current code, but it does not fit in the proposed changes of https://github.com/lit/lit/pull/3128/files

Reply:
I understand the need for it to be close to the logic. But it's too much happening in an `if` statement. It makes it more difficult to read and maintain later on. Accounting for a fallback at the module level is more durable. This is technically a nested if statement except it's nested at the condition statement.

We are `if`-ing a prototype and a ternary inside a conditional statement does not read that way to me 

Reply:
> element and HTMLElement are not the interchangeable part here.

`HTMLElement` is not used in #3128, this is what i mean by `conflicts`. Not conflicts as in git i mean conflicts as in implementation and correctness

Reply:
Was lurking and not very plugged into this, but was reading the source on the branch and:

> it's too much happening in an if statement

Agree with Brian on this one. If you're keeping this logic, is there a way to rewrite this without hurting perf? Nested `if` I think would be okay here if only for readability

Reply:
Yup. Now with the reference to `HTMLElement` moved to a lifecycle that won't be called on the server, this gnarly condition here won't be necessary. To address prop/attribute handling during server-render, I anticipate it'll be a much more straight forward single `if (NODE_MODE)` or `if (isServer)` check instead.